Paintroid/src/main/java/org/catrobat/paintroid/command/implementation/DefaultCommandManager.kt

```diff
@@ -175,10 +175,10 @@ class DefaultCommandManager(
     private fun backUpCheckBoxes(layerCount: Int, checkBoxes: MutableList<Boolean>) {
         if (layerCount > 1) {
             for (index in layerCount - 1 downTo 0) {
-                checkBoxes[index] = layerModel.getLayerAt(index).checkBox
+                checkBoxes[index] = layerModel.getLayerAt(index)!!.checkBox
```

Comment:
`!!` should never be used in production code, because it can result in unnecessary crashes.
Instead you can write something like this `checkBoxes[index] = layerModel.getLayerAt(index)?.checkBox == true`

Reply:
i now added a way to "recover" the undo/redo stack

Reply:
I now added a way to restore the case of a failure undoing the addLayerCommand and mergeLayerCommand.
AddLayerCommand gets restored completely, mergeLayerCommand should work as intented, if concurrency-issues occure, a layer which should be hidden COULD be set visible, due to default values being set in an error case.
Hope you think as well, that this is sufficient.

Reply:
The changes look good :)